helix-term/src/ui/editor.rs

```diff
@@ -604,22 +604,27 @@ impl EditorView {
             use helix_core::match_brackets;
             let pos = doc.selection(view.id).primary().cursor(text);
 
-            let pos = match_brackets::find_matching_bracket(syntax, doc.text(), pos)
-                .and_then(|pos| view.screen_coords_at_pos(doc, text, pos));
-
+            let pos = match_brackets::find_matching_bracket(syntax, doc.text(), pos);
             if let Some(pos) = pos {
-                // ensure col is on screen
-                if (pos.col as u16) < viewport.width + view.offset.col as u16
-                    && pos.col >= view.offset.col
-                {
-                    let style = theme.try_get("ui.cursor.match").unwrap_or_else(|| {
-                        Style::default()
-                            .add_modifier(Modifier::REVERSED)
-                            .add_modifier(Modifier::DIM)
-                    });
-
-                    surface[(viewport.x + pos.col as u16, viewport.y + pos.row as u16)]
-                        .set_style(style);
+                // Don't render over selection
+                if !doc.selection(view.id).primary().contains(pos) {
+                    let pos = view.screen_coords_at_pos(doc, text, pos);
+
+                    if let Some(pos) = pos {
+                        // ensure col is on screen
+                        if (pos.col as u16) < viewport.width + view.offset.col as u16
+                            && pos.col >= view.offset.col
+                        {
+                            let style = theme.try_get("ui.cursor.match").unwrap_or_else(|| {
+                                Style::default()
+                                    .add_modifier(Modifier::REVERSED)
+                                    .add_modifier(Modifier::DIM)
+                            });
+
+                            surface[(viewport.x + pos.col as u16, viewport.y + pos.row as u16)]
+                                .set_style(style);
+                        }
+                    }
```

Comment:
There are a few nested `if let Some`s here that can be refactored to reduce the nesting:

```rs
if let Some(pos) = match_brackets::find_matching_bracket(syntax, doc.text(), pos)
    // Don't render over the primary selection.
    .filter(|pos| !doc.selection(view.id).primary().contains(*pos))
    .and_then(|pos| view.screen_coords_at_pos(doc, text, pos))
{
    if (pos.col as u16) < viewport.width + view.offset.col as u16
        && pos.col >= view.offset.col
    {
        let style = theme.try_get("ui.cursor.match").unwrap_or_else(|| {
            Style::default()
                .add_modifier(Modifier::REVERSED)
                .add_modifier(Modifier::DIM)
        });

        surface[(viewport.x + pos.col as u16, viewport.y + pos.row as u16)]
            .set_style(style);
    }
}
```

Reply:
Applied the suggested changes